target/linux/pistachio/base-files/etc/init.d/set_mac

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh /etc/rc.common
+
+. /lib/functions/mac.sh
+
+START=90
+STOP=15
+
+start()
+{
+    generate_mac $SOFTAP_OTP_REG0
+    uci set wireless.ap.macaddr=$MAC
+    uci commit wireless
+    sync
+    /etc/init.d/network restart
+
+    generate_mac $ETH_OTP_REG0
+    ifconfig eth0 down
+    ifconfig eth0 hw ether $MAC
+    ifconfig eth0 up
+}
```

Comment:
please do not use these hacks to set the mac address, have a look at how it is solved here: target/linux/lantiq/base-files/etc/board.d/02_network